components/calendar/generateCalendar.tsx

```diff
@@ -198,10 +201,14 @@ function generateCalendar<DateType>(generateConfig: GenerateConfig<DateType>) {
       triggerPanelChange(mergedValue, newMode);
     };
 
-    const onInternalSelect = (date: DateType) => {
+    const onInternalSelect = (date: DateType, selectType?: CalendarSelectable) => {
       triggerChange(date);
 
-      onSelect?.(date);
+      if (selectable && selectable.length > 0 && selectType) {
+        if (!selectable.includes(selectType)) return;
```

Comment:
When `null` or `[]`, can all be selectable?

Reply:
This should check `selectType` only if `selectable` is set, otherwise, it works as it currently does (any change in date calls `onSelect`).

Reply:
So if `selectable` = `null` | `undefined` | `[]`, no check is done and the function works as it does now .

Reply:
Yeah, I see that code now, Thanks.